### PR TITLE
Gh-278: Update image tagging to work with new versioning

### LINKED ISSUE #278: Update image tagging to work with new versioning
The changes in #275 add the Accumulo version to the version tag for certain images. The GitHub Actions workflow requires updating to work correctly with this new tag format.

This is also a good opportunity to upgrade the workflows to make greater use of Actions instead of scripts and to also push images to GHCR.

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
Also improve regex in the script to use extended regex

# Related Issue

- Resolve #278

### CODE REVIEW

File: cd/publish_images.sh [hunk 1]
```diff
@@ -51,16 +83,18 @@ source "${ROOT_DIR}"/docker/gaffer-pyspark-notebook/.env
...
-pushTags gchq/gaffer-ui "${GAFFER_TOOLS_VERSION}" "${APP_VERSION}"
-pushTags gchq/gaffer-road-traffic-loader "${GAFFER_VERSION}" "${APP_VERSION}"
-pushTags gchq/gaffer-pyspark-notebook "${GAFFER_VERSION}" "${APP_VERSION}"
-pushTags gchq/gaffer-jhub-options-server "${JHUB_OPTIONS_SERVER_VERSION}" "${APP_VERSION}"
-pushTags gchq/spark-py "${SPARK_VERSION}" "${APP_VERSION}"
+# Push to Container Repositories
+pushContainer gchq/hdfs "${HADOOP_VERSION}"
+pushContainer gchq/accumulo "${ACCUMULO_VERSION}"
+pushContainer gchq/gaffer "${GAFFER_VERSION}-accumulo-${ACCUMULO_VERSION}"
+pushContainer gchq/gaffer-rest "${GAFFER_VERSION}"
```
Comment (t92549) [inline cd/publish_images.sh:95]: I think gaffer-rest needs `ACCUMULO_VERSION` too

File: cd/publish_images.sh [hunk 2]
```diff
@@ -16,27 +16,59 @@
...
+    fi
+}
+
+# Generate tags for Gaffer image based on version
+getGafferTags() {
+    if [[ $version =~ .*alpha.* ]];
+    then
+      tags="${name}:${version} ${name}:latest"
+    else
+      tags="$(echo ${version} | sed -E "s|([0-9]+)\.([0-9]+)\.([0-9]+)-accumulo-([0-9.]*)|${name}:${version} ${name}:\1.\2.\3 ${name}:\1.\2 ${name}:\1 ${name}:latest|")"
```
Comment (t92549) [inline cd/publish_images.sh:45]: What does this mean? `${name}:\1.\2.\3`

File: cd/publish_images.sh (hunk 1 above)
Comment (GCHQDeveloper314) [reply cd/publish_images.sh:95]: Yes it does, missed that. Will fix it now.

File: cd/publish_images.sh (hunk 2 above)
Comment (GCHQDeveloper314) [reply cd/publish_images.sh:45]: There's 3 capture groups, one for each part of the version. First group is Major, then Minor, then Patch. The other regex doesn't need this as the `$version` is already in this format and can be used directly.

**Review by t92549 — APPROVED** [review]

**Pull request merged**